crates/l1tx/src/messages.rs

```diff
@@ -45,6 +46,21 @@ impl ProtocolTxEntry {
     }
 }
 
+/// Da data retrieved from L1 transaction.
+#[derive(Clone, Debug)]
+pub struct DaEntry {
+    #[allow(unused)]
+    commitment: Buf32,
+    #[allow(unused)]
+    blob: Vec<u8>,
+}
+
+impl DaEntry {
+    pub fn new(commitment: Buf32, blob: Vec<u8>) -> Self {
+        Self { commitment, blob }
```

Comment:
It doesn't *really* make sense to take both of these args separately, since the commitment can be computed from the blob.

Reply:
Yes, but shis prevents re-computation of the commitment where it is already computed at the caller.

Reply:
In contexts where we want the blob contents already it's cheap to hash it.  At least change this to `new_unchecked` to signal that it doesn't validate that they match.